code/game/objects/items/weapons/cards_ids.dm

```diff
@@ -370,262 +388,238 @@
+	return data
+
+/obj/item/card/id/syndicate/ui_interact(mob/user, datum/tgui/ui = null)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "AgentCard", name)
+		ui.open()
+
+/obj/item/card/id/syndicate/attack_self(mob/user)
+	if(!ishuman(user))
+		return
+	if(!registered_human)
+		registered_human = user
+	if(registered_human != user)
+		flash_card(user)
+		return
+	switch(tgui_alert(user, "Would you like to display [src] or edit it?", "Choose", list("Edit", "Show")))
+		if("Show")
+			flash_card(user)
+		if("Edit")
+			ui_interact(user)
+
+/obj/item/card/id/syndicate/proc/delete_info(datum/tgui/ui)
+	name = initial(name)
+	registered_name = initial(registered_name)
+	icon_state = initial(icon_state)
+	sex = initial(sex)
+	age = initial(age)
+	assignment = initial(assignment)
+	rank = initial(rank)
+	associated_account_number = initial(associated_account_number)
+	blood_type = initial(blood_type)
+	dna_hash = initial(dna_hash)
+	fingerprint_hash = initial(fingerprint_hash)
+	photo = null
+	registered_human.sec_hud_set_ID()
+	registered_human = null
+	ui.close()
+
+/obj/item/card/id/syndicate/proc/clear_access()
+	access = initial_access.Copy() // Initial() doesn't work on lists
+	to_chat(registered_human, "<span class='notice'>Card access reset.</span>")
+
+/obj/item/card/id/syndicate/proc/change_ai_tracking()
+	untrackable = !untrackable
+
+/obj/item/card/id/syndicate/proc/change_name(option, name)
+	var/new_name
+	if(option == "Primary")
+		new_name = ishuman(registered_human) ? registered_human.real_name : registered_human.name
+	else if(option == "Secondary")
+		new_name = tgui_input_list(registered_human, "Whose name do you want to copy?", "Agent ID - Name", GLOB.crew_list)
+		if(isnull(new_name))
 			return
-		src.registered_name = t
+	else
+		new_name = name
+
+	registered_name = reject_bad_name(new_name, TRUE)
+	UpdateName()
+	to_chat(registered_human, "ID name has been changed to [new_name].")
 
-		var/u = tgui_input_text(user, "What occupation would you like to put on this card?\nNote: This will not grant any access levels other than maintenance.", "Agent Card Job Assignment", "Agent", MAX_MESSAGE_LEN)
-		if(!u)
-			to_chat(user, "<span class='warning'>Invalid assignment.</span>")
-			src.registered_name = ""
+/obj/item/card/id/syndicate/proc/change_photo()
+	if(!COOLDOWN_FINISHED(src, new_photo_cooldown))
+		return
+	var/job_clothes = null
+	if(assignment)
+		job_clothes = assignment
+	var/icon/newphoto = get_id_photo(registered_human, job_clothes)
+	if(!newphoto)
+		return
+	photo = newphoto
+	COOLDOWN_START(src, new_photo_cooldown, 10 SECONDS) // This proc is expensive, we don't want people spamming it.
+
+/obj/item/card/id/syndicate/proc/change_appearance(new_appearance)
+	if(!new_appearance)
+		return
+	if(new_appearance in icon_states(icon))
+		icon_state = new_appearance
+
+/obj/item/card/id/syndicate/proc/change_sex(new_sex)
+	if(!Adjacent(registered_human) || isnull(new_sex))
+		return
+	sex = new_sex
```

Comment:
This needs some kind of sanitization. Otherwise people can put whatever strings they like in here (including HTML tags)